src/main/java/org/dsa/iot/knx/KnxIPTunnelingConnection.java

```diff
@@ -40,12 +42,26 @@ public void init() {
 	}
 
 	@Override
+	protected void connect() {
+		super.connect();
+		if (null != networkLink && null != communicator) {
+			generateGatewayNode();
+		}
+
+	}
+
+	private void generateGatewayNode() {
+		Node child = node.createChild(ATTR_GATEWAY_NAME).build();
+		child.setAttribute(ATTR_DEVICE_ADDRESS, new Value(deviceAddress));
+		DeviceNode deviceNode = new DeviceNode(getConnection(), null, child);
```

Comment:
Device folder could be a tree. For the top node, no need to pass a valid root node because its root is itself.